From 5a5629f76669c7bac25c84067608c72ad2862fa2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Ol=C5=A1=C3=A1k?= <marek.olsak@amd.com>
Date: Thu, 5 Oct 2023 06:37:22 -0400
Subject: [PATCH 475/834] ac/gpu_info: set gfx and compute IB padding to only 8
 dwords

This is what the kernel reports and what PAL seems to be doing.

Reviewed-by: Samuel Pitoiset <samuel.pitoiset@gmail.com>
Reviewed-by: Pierre-Eric Pelloux-Prayer <pierre-eric.pelloux-prayer@amd.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25578>
---
 src/amd/common/ac_gpu_info.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 0d12d28ea3f..8029cfa71b2 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -688,8 +688,8 @@ bool ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    }
 
    /* Set dword padding minus 1. */
-   info->ip[AMD_IP_GFX].ib_pad_dw_mask = 0x3f;
-   info->ip[AMD_IP_COMPUTE].ib_pad_dw_mask = 0x3f;
+   info->ip[AMD_IP_GFX].ib_pad_dw_mask = 0x7;
+   info->ip[AMD_IP_COMPUTE].ib_pad_dw_mask = 0x7;
    info->ip[AMD_IP_SDMA].ib_pad_dw_mask = 0xf;
    info->ip[AMD_IP_UVD].ib_pad_dw_mask = 0xf;
    info->ip[AMD_IP_VCE].ib_pad_dw_mask = 0x3f;
-- 
2.42.0

